Fine-tuned the style of contribution grid, which is exactly the same as github

xiandanin opened the pull request:
Added __rounded corners__, __border__, __corrected colors__, and the parameters are completely referenced to the latest github, making it closer to the style of github.

![image](https://user-images.githubusercontent.com/17588779/95735488-736cf100-0cb7-11eb-822b-7f131ea3dc86.png)

sallar commented on __tests__/draw.test.js:
```diff
@@ -14,7 +16,13 @@ test("draws contributions correctly", () => {
   });
   expect(canvas.toDataURL()).toMatchSnapshot();
+
+  /*
```
this comment can be removed

sallar commented on __tests__/draw.test.js:
```diff
@@ -2,7 +2,9 @@ import path from "path";
 import { createCanvas, registerFont } from "canvas";
+
 const canvas = createCanvas(200, 200);
+import fs from "fs";
```
unnecessary import

sallar commented on __tests__/draw.test.js:
```diff
@@ -14,7 +16,13 @@ test("draws contributions correctly", () => {
     themeName: "__test__",
-    fontFace: "PlexMono"
+    fontFace: "PlexMono",
+    scaleFactor: 2
```
this should be dynamic

sallar commented on src/themes.js:
```diff
@@ -1,22 +1,24 @@
-    grade2: "#7bc96f",
-    grade1: "#c6e48b",
+    text: "#24292e",
+    meta: "#767676",
```
the test text colors were white before the text rendering isn't the same on the CI vs the dev machine, so they need to be white so the tests wont fail

sallar commented on src/themes.js:
```diff
@@ -1,22 +1,24 @@
-    grade1: "#c6e48b",
+    text: "#24292e",
+    meta: "#767676",
+    stroke:'rgba(27,31,35,.04)',
```
the other themes dont have strokes... what happens?

sallar commented on src/index.js:
```diff
@@ -112,10 +167,12 @@ function drawYear(ctx, opts = {}) {
+      }
+
+      ctx.fillStyle = theme[`grade${day.info.intensity}`];
+      ctx.strokeStyle = theme["stroke"];
```
Since the other themes dont have strokes this fails

sallar commented on src/index.js:
```diff
@@ -161,11 +219,14 @@ function drawMetaData(ctx, opts = {}) {
   for (let x = 0; x < 5; x += 1) {
     ctx.fillStyle = theme[`grade${x}`];
-    ctx.fillRect(width - canvasMargin - (boxWidth + boxMargin) * (themeGrades) - 27,textHeight + boxWidth,10,10);
+    ctx.strokeStyle = theme["stroke"];
```
also this

sallar reviewed (changes requested):
Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.

But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

xiandanin commented:
> Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.
> 
> But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

I made some modifications as you suggested:
* I added the "newGithub" theme to show that this is the latest official github style.
* Added "radius" and "stroke" parameters, but this is optional. If the theme does not contain this parameter, it will work normally.
* Change the text color of "__test__" theme to white.
* "scaleFactor" can still be passed in as a parameter, but it is dynamic by default.

xiandanin reviewed (commented):
> Thanks for your PR. This is a great idea. However there are a few things that need to be fixed that I have commented on the PR.
> 
> But also, I think this needs to be a "theme option", for example the classic github theme didnt have rounded corners, etc. Or maybe an option on the whole package, that the user can control.

I made some modifications as you suggested:
* I added the "newGithub" theme to show that this is the latest official github style.
* Added "radius" and "stroke" parameters, but this is optional. If the theme does not contain this parameter, it will work normally.
* Change the text color of "__test__" theme to white.
* "scaleFactor" can still be passed in as a parameter, but it is dynamic by default.

sallar commented:
I apologize for not merging this yet @xiandanin - I need to get a chance to sit down and work on this.
